config/core/deployments/autoscaler.yaml

```diff
@@ -102,3 +112,4 @@ spec:
     targetPort: 8080
   selector:
     app: autoscaler
+  clusterIP: None
```

Comment:
This is a breaking change as `clusterIP` is immutable. The existing autoscaler service need to be deleted before installing new knative version with this change.

@mattmoor @markusthoemmes @vagababov  How do we usually handle this kind of change?

Reply:
Is this even correct? How will the activator talk to the autoscalers?

Reply:
Autoscaler DNS is automatically configured by the pods matched by `selector`. See https://kubernetes.io/docs/concepts/services-networking/service/#with-selectors

This change is required to make statefulset pod DNS work: https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#stable-network-id

Reply:
We can't make such a breaking change, so we should discuss options.  Why can't this be a new parallel service, and we delete the old service once we have migrated stuff over?

Reply:
Yeah, that was my idea as well, when I talked to Yanwei.

Reply:
A different service should work. However people may depend on the service name to get the autoscaler metrics which are exposed in 9090 port. Do we guarantee that knative service names won't change?

Reply:
Changed to a new service called `autoscaler-ss`.